public/apps/configuration/panels/permission-list.tsx

```diff
@@ -0,0 +1,242 @@
+    <EuiContextMenuItem
+      key="blank"
+      // TODO create from selection
+      onClick={() => {}}
+      disabled={selection.length === 0}
+    >
+      Create from selection
+    </EuiContextMenuItem>,
+  ];
+
+  const actionsButton = (
+    <EuiButton
+      iconType="arrowDown"
+      iconSide="right"
+      onClick={() => {
+        setActionsPopoverOpen((prevState) => !prevState);
+      }}
+    >
+      Actions
+    </EuiButton>
+  );
+
+  const createActionGroupMenuButton = (
+    <EuiButton
+      iconType="arrowDown"
+      iconSide="right"
+      onClick={() => {
+        setCreateActionGroupPopoverOpen((prevState) => !prevState);
+      }}
+      fill
+    >
+      Create action group
+    </EuiButton>
+  );
+
+  return (
+    <>
+      <EuiPageHeader>
+        <EuiTitle size="l">
+          <h1>Permission</h1>
+        </EuiTitle>
+      </EuiPageHeader>
+      <EuiPageContent>
+        <EuiPageContentHeader>
+          <EuiPageContentHeaderSection>
+            <EuiTitle size="s">
+              <h3>Permissions ({actionGroups.length})</h3>
+            </EuiTitle>
+            <EuiText size="xs" color="subdued">
+              Permissions defines type of access to the cluster or the specified indices. An action
+              group is a set of predefined single permissions and/or another set of action groups.
+              You can often achieve your desired security posture using some combination of the
+              default action groups. Or you can create your own group.{' '}
+              <EuiLink external={true} href="/">
+                Learn More
+              </EuiLink>
+            </EuiText>
+          </EuiPageContentHeaderSection>
+          <EuiPageContentHeaderSection>
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <EuiPopover
+                  id="actionsMenu"
+                  button={actionsButton}
+                  isOpen={isActionsPopoverOpen}
+                  closePopover={() => {
+                    setActionsPopoverOpen(false);
+                  }}
+                  panelPaddingSize="s"
+                >
+                  <EuiContextMenuPanel items={actionsMenuItems} />
+                </EuiPopover>
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiPopover
+                  id="actionsMenu"
+                  button={createActionGroupMenuButton}
+                  isOpen={isCreateActionGroupPopoverOpen}
+                  closePopover={() => {
+                    setActionsPopoverOpen(false);
+                  }}
+                  panelPaddingSize="s"
+                >
+                  <EuiContextMenuPanel items={createActionGroupMenuItems} />
+                </EuiPopover>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiPageContentHeaderSection>
+        </EuiPageContentHeader>
+        <EuiPageBody>
+          <EuiInMemoryTable
+            loading={actionGroups === [] && !errorFlag}
```

Comment:
Curious what if the returned actionGroups is indeed empty?

Reply:
Then it shows nothing.

Reply:
Will the loading be always true then?

Reply:
This should be an abnormal case since there are a set of reserved actionGroups that not allows to be deleted.

Reply:
Yes that will shows infinite loading.